File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int low = 0,high = nums.size()-1;
        int mini = INT_MAX;
        while(low<=high){
            int mid = low+(high-low)/2;
            mini = min(nums[mid],mini);
            if(nums[low]<=nums[high]){
                mini = min(mini,nums[low]);
            }
            if(nums[low]<nums[mid]){
                mini = min(nums[low],mini);
                low = mid+1;
            }
            else {
                mini = min(nums[high],mini);
                high = mid-1;
                }
        }return mini;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        r = len(nums)-1
        l = 0
        while(l<r):
            mid = l+(r-l)//2
            if nums[mid]<nums[r]:
                r = mid
            else:
                l=mid+1

        return nums[l]